front/components/data_source_view/DataSourceViewSelector.tsx

```diff
@@ -30,30 +37,147 @@ import {
   isManaged,
   isWebsite,
 } from "@app/lib/data_sources";
+import { useVaults } from "@app/lib/swr/vaults";
+import { classNames } from "@app/lib/utils";
+import { getVaultIcon, getVaultName } from "@app/lib/vaults";
 
 const MIN_TOTAL_DATA_SOURCES_TO_GROUP = 12;
 const MIN_DATA_SOURCES_PER_KIND_TO_GROUP = 3;
+const ONLY_ONE_VAULT_PER_SELECTION = true;
 
 interface DataSourceViewsSelectorProps {
   owner: LightWorkspaceType;
   dataSourceViews: DataSourceViewType[];
+  allowedVaults?: VaultType[];
   selectionConfigurations: DataSourceViewSelectionConfigurations;
   setSelectionConfigurations: Dispatch<
     SetStateAction<DataSourceViewSelectionConfigurations>
   >;
   viewType: ContentNodesViewType;
 }
 
+function VaultSelector({
+  owner,
+  dataSourceViews,
+  allowedVaults,
+  selectionConfigurations,
+  setSelectionConfigurations,
+  viewType,
+}: DataSourceViewsSelectorProps) {
+  const dataSourceViewsByVaultId = useMemo(
+    () => _.groupBy(dataSourceViews, (dsv) => dsv.vaultId),
+    [dataSourceViews]
+  );
+
+  const { vaults, isVaultsError, isVaultsLoading } = useVaults({
+    workspaceId: owner.sId,
+  });
+
+  const [selectedVault, setSelectedVault] = useState<string>(
+    Object.keys(selectionConfigurations).length > 0
+      ? selectionConfigurations[Object.keys(selectionConfigurations)[0]]
+          .dataSourceView.vaultId
+      : ""
+  );
```

Comment:
```suggestion
const [selectedVault, setSelectedVault] = useState<string>(() => {
  const firstKey = Object.keys(selectionConfigurations)[0];
  return firstKey ? selectionConfigurations[firstKey].dataSourceView.vaultId : "";
});
```

Reply:
I think you still need to check the length but I'll update to improve lisibility.